### PR TITLE
[core] Supress infeasible warning

### LINKED ISSUE #14051: Toggling the resource "pending" warning to improve library UX
### Problem

It's often easier to 'fire and forget' for starting actors when resources are not available (and waiting for Ray to autoscale). However, we often see this nefarious, notorious error message:

```python
2020-01-13 18:15:09,447	WARNING worker.py:1062 -- The actor or task with ID ffffffffffffffff45b95b1c0100 is pending and cannot currently be scheduled. It requires {CPU: 1.000000} for execution and {CPU: 1.000000} for placement, but this node only has remaining {node:172.22.225.108: 1.000000}, {CPU: 2.000000}, {memory: 16.113281 GiB}, {GPU: 1.000000}, {object_store_memory: 5.566406 GiB}. In total there are 0 pending tasks and 2 pending actors on this node. This is likely due to all cluster resources being claimed by actors. To resolve the issue, consider creating fewer actors or increase the resources available to this Ray cluster. You can ignore this message if this Ray cluster is expected to auto-scale.
```

You might say "ok well you can solve that with placement groups". Unfortunately, this introduces complexity in various ways:

### User ends up needing to manage the creation of the placement group

Ideally, you have this:

```python
actors = []
# start N actors
for i in range(actors):
    new_pg.start()
    actor.options(pg).remote() 
    actors += [actor]
```
But in order to avoid the warning message: 

```python
# start N actors
for i in range(actors):
    groups += [(new_pg.start(), new_pg.ready())]
    
while pg in ray.wait(groups):
    actor.options(pg).remote()
    actors += [actor]
```

### If you want to reuse the placement group, you have to introduce a new state in your application level scheduling

Ideal:
```python
# PG reuse
for old_actor, _ in ray.wait(actors):
	old_actor.stop.remote()
    pg = old_actor.get_pg()
    actors += [new_actor.options(pg).remote()]
```

But in order to avoid the warning message: 

```python
# PG reuse
stopping_actors = []
while True:
	actor, _ = ray.wait(actors, timeout=0.1)
    if actor:
        stopping_actors.append(actor.stop.remote())
    if stopping_actors:
        old_actor, _ = ray.wait(stopping_actors, timeout=0.1)
        if old_actor:
            pg = old_actor.get_pg()
            actors += [new_actor.options(pg).remote()]
```

## Proposed Solution

Can we just have a configurable warning option?

```python
Actor.options(warn_about_placement_delay_s=60).remote
```

cc @ericl @edoakes @wuisawesome @krfricke

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?
This warning can have a tendency to be a little spammy. Especially in an autoscaling cluster, it's very reasonable to want to avoid this warning entirely. 

## Related issue number
Closes #14051 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
I thought the original issue was to have it per task, not globally?

**Comment by wuisawesome** [comment]
oh lol in hindsight that's probably what he meant... i lost track of the alternatives i guess. 

i can make a PR with that later. much more plumbing involved...

**Comment by robertnishihara** [comment]
@richardliaw @wuisawesome Is this intended to be part of the API or is this just an internal hack that we'll use in the short term until we do it automatically (if the latter, it should be called `_internal_supress_infeasible_warning`)? I'd really prefer to not have this flag, and it seems like there are other good options.

**Comment by richardliaw** [comment]
@wuisawesome I'm going to convert this to a draft PR (as previously suggested)

**Review by richardliaw — DISMISSED** [review]

**Comment by richardliaw** [comment]
Unfortunately a test is failing?

**Comment by richardliaw** [comment]
Can you also add a test for this behavior (to prevent regressions)?

File: doc/source/placement-group.rst [hunk 1]
```diff
@@ -162,6 +162,12 @@ Let's create a placement group. Recall that each bundle is a collection of resou
   # Wait until placement group is created.
   ray.get(pg.ready())
 
+ .. note::
+
+    When using placement groups, users should ensure that their tasks are ready (by calling ``ray.get(pg.ready())``)
```
Comment (pcmoritz) [inline doc/source/placement-group.rst:167]: `tasks are ready` -> shouldn't this be `placement groups are ready/scheduled`?

File: python/ray/util/placement_group.py [hunk 2]
```diff
@@ -109,9 +109,7 @@ def bundle_count(self):
         return len(self.bundle_cache)
 
     def _get_none_zero_resource(self, bundle: List[Dict]):
-        # This number shouldn't be changed.
-        # When it is specified, node manager won't warn about infeasible
-        # tasks.
+        # Any number between 0-1 should be fine.
         INFEASIBLE_TASK_SUPPRESS_MAGIC_NUMBER = 0.0101
```
Comment (edoakes) [inline python/ray/util/placement_group.py:113]: Looks like this isn't used anymore after this change -- can you please remove it from here and `node_manager.cc` if that's the case?

Comment (edoakes) [reply python/ray/util/placement_group.py:113]: ah yeah LGTM. let's rename it to just magic number or something, the name is pretty confusing now that it doesn't actually suppress anything

**Review by ericl — APPROVED** [review]

**Pull request merged**